### PR TITLE
chore: v6 migration guide

### PR DESCRIPTION
**Pull request by sighphyre**

### CODE REVIEW

File: v6_MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Migrating to Unleash-Client-Python 6.0.0
+
+This guide highlights the key changes you should be aware of when upgrading to v6.0.0 of the Unleash client.
+
+## Direct access to feature toggles
```
Comment (melindafekete) [inline v6_MIGRATION_GUIDE.md:5]:
```suggestion
## Removed direct access to feature flags
```

File: v6_MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+Direct access to the feature toggle objects through `UnleashClient.features` has been removed. All classes related to the internal representation of feature toggles are no longer publicly accessible in the SDK.
```
Comment (melindafekete) [inline v6_MIGRATION_GUIDE.md:7]:
```suggestion
Direct access to the feature flag objects through `UnleashClient.features` has been removed. All classes related to the internal representation of feature flags are no longer publicly accessible in the SDK.
```

File: v6_MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+The SDK now provides an `UnleashClient.feature_definitions()` method, which returns a list of feature toggle names, their type, and the project they're bound to.
```
Comment (melindafekete) [inline v6_MIGRATION_GUIDE.md:9]:
```suggestion
The SDK now provides an `UnleashClient.feature_definitions()` method, which returns a list of feature flag names, their type, and the project they're bound to.
```

File: v6_MIGRATION_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+## Changes to custom strategies
+
+Custom strategies have undergone some changes that require updates to their implementations. This is a strict requirement: any strategy that does not implement the correct interface will throw an exception at startup.
```
Comment (melindafekete) [inline v6_MIGRATION_GUIDE.md:13]:
```suggestion
Custom strategies have undergone some changes that require updates to their implementations. This is a strict requirement; any strategy that does not implement the correct interface will throw an exception at startup.
```

File: v6_MIGRATION_GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+The interface changes are as follows:
+
+- Strategies no longer inherit from a base class.
+- The apply method now accepts a second parameter, parameters. In legacy versions, this functionality was managed by the load_provisioning method.
```
Comment (melindafekete) [inline v6_MIGRATION_GUIDE.md:18]:
```suggestion
- The apply method now accepts a second parameter, `parameters`. In legacy versions, this functionality was managed by the `load_provisioning()` method.
```

File: v6_MIGRATION_GUIDE.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+This is now written as:
+
+``` python
+class CatStrategy:
+    def apply(self, parameters: dict, context: dict = None) -> bool:
+        default_value = False
+
+        parameters = [x.strip() for x in parameters["sound"].split(",")]
```
Comment (melindafekete) [inline v6_MIGRATION_GUIDE.md:43]: Since the argument is also called `parameters`, should we call this something different?

**Review by melindafekete — APPROVED** [review]

File: v6_MIGRATION_GUIDE.md (hunk 6 above)
Comment (sighphyre) [reply v6_MIGRATION_GUIDE.md]: Guess I spend too much time in Rust these days, the shadow feels completely normal to me haha

Sure. I've swapped that out for `parsed_parameters`

**Pull request merged**